### ISSUE TITLE
caching is broken in the pipeline

### ISSUE THREAD

**Issue by ksaur** [microsoft/hummingbird#506]
We keep hitting
```
Unable to reserve cache with key Linux-1-tvm-0.7, another job may be creating this cache.
```
which means our pipeline takes >40min each run.

It seems there is an [open issue.](https://github.com/actions/toolkit/issues/658)

Maybe we can try renaming it as a workaround?

**Comment by ksaur** [comment]
NOPE.  DENIED. (from [https://github.com/microsoft/hummingbird/runs/2460962168](pipeline))
```
Post job cleanup.
Unable to reserve cache with key Linux-2-tvm-0.7, another job may be creating this cache.
```

**Comment by interesaaat** [comment]
Let's try with 3?

**Comment by ksaur** [comment]
Now we have a broken build so fun!

```
Post job cleanup.
Warning: Cache already exists. Scope: refs/pull/509/merge, Key: macOS-3-tvm-0.7, Version: (null)
```
this was at the end of my [PR run](https://github.com/microsoft/hummingbird/pull/509/checks?check_run_id=2462770958), and now we have the same issue as before on `main` branch ([run](https://github.com/microsoft/hummingbird/runs/2463097691)):


```
Run actions/cache@v1
Cache Size: ~51 MB (53133296 B)
/usr/bin/tar -xz -f /Users/runner/work/_temp/672e2837-6973-456c-a6fa-b9f648e5562c/cache.tgz -C /Users/runner/incubator-tvm
Cache restored from key: macOS-3-tvm-0.7

...
Run python setup.py install
Traceback (most recent call last):
  File "setup.py", line 61, in <module>
    LIB_LIST, __version__ = get_lib_path()
  File "setup.py", line 49, in get_lib_path
    lib_path = libinfo["find_lib_path"]()
  File "./tvm/_ffi/libinfo.py", line 141, in find_lib_path
    raise RuntimeError(message)
RuntimeError: Cannot find the files.
```

**Comment by interesaaat** [comment]
Should we do with the python version as you were suggesting?

**Comment by ksaur** [comment]
mmmkay it says

```
Post job cleanup.
/bin/tar -cz -f /home/runner/work/_temp/e35fce1f-3f92-4662-98d6-814decce9db6/cache.tgz -C /home/runner/incubator-tvm .
Cache saved successfully
```

let's see.   i'll rerun the same

**Comment by ksaur** [comment]
closed with #510 . i really hope it keeps working!
